app/routes.py

```diff
@@ -1 +1,240 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+from datetime import date
+from sqlalchemy import select, desc, asc
+import requests
+import os
+from dotenv import load_dotenv
+load_dotenv()
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_task(task_id):
+    '''Takes in one task_id, returns the Task object if it exists, otherwise aborts.'''
+    try:
+        task_id = int(task_id)
+    except:
+        abort(make_response({"message": f"task {task_id} invalid, id must be integer"}, 400))
+    
+    task = Task.query.get(task_id)
+
+    if not task:
+        abort(make_response({"message": f"task {task_id} not found"}, 404))
+
+    return task
+
+def check_is_complete(task):
+    '''Takes in one instance of task object, returns the value that should be assigned to is_complete for that object.'''
+    if task.completed_at is None:
+        return False
+    return True
```

Comment:
This can be simplified to:

```suggestion
    return bool(task.completed_at)
```